PR Add Feature to Retrieve Non-Leaf Bounding Boxes

cbergoon opened the pull request:
Would you consider accepting this PR which adds functionality to get the bounding boxes of all non-leaf nodes? This is useful for visualizations of the tree and analyzing resulting clusters of data. I have implemented the feature and tests on PR. 

This change introduces one public method ```GetAllBoundingBox``` on ```Rtree``` which calls ```getAllBoundingBox```. ```getAllBoundingBox``` traverses the tree from root building a slice of non-leaf bounding boxes and returns it.

dhconnelly commented on rtree.go:
```diff
@@ -486,6 +501,16 @@ func (tree *Rtree) NearestNeighbor(p Point) Spatial {
 
+// GetAllBoundingBox returning slice of bounding boxes by traversing tree. Slice
+// includes bounding boxes from all non-leaf nodes.
+func (tree *Rtree) GetAllBoundingBox() []*Rect {
```
could you rename to GetAllBoundingBoxes (and the helper function to getAllBoundingBoxes)?

cbergoon replied:
I renamed theses functions.

dhconnelly commented:
Looks good! I left one comment. I also just merged another pull request, so it looks like you'll need to resolve conflicts in rtree_test.go.

cbergoon commented:
I renamed those functions, fixed an obvious bug in my logic, and resolved the merge conflicts. 

Let me know if you'd like to see anything else. 

-Cam
